extensions/ql-vscode/src/authentication.ts

```diff
@@ -0,0 +1,60 @@
+import * as vscode from 'vscode';
+import * as Octokit from '@octokit/rest';
+
+const GITHUB_AUTH_PROVIDER_ID = 'github';
+
+// 'repo' scope should be enough for triggering workflows. For a comprenhensive list, see:
+// https://docs.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps
+const SCOPES = ['repo'];
+
+/** 
+ * Handles authentication to GitHub, using the VS Code [authentication API](https://code.visualstudio.com/api/references/vscode-api#authentication).
+ */
+export class Credentials {
+  private octokit: Octokit.Octokit | undefined;
+
+  async initialize(context: vscode.ExtensionContext): Promise<void> {
+    this.registerListeners(context);
+    this.setOctokit();
```

Comment:
This line needs an `await`.

You may find it useful to add the `"@typescript-eslint/no-floating-promises": "error"` linting rule, though I tried it and there are a lot of cases caused by async logging calls, so probably not something to look at in this PR.

Reply:
That's a good idea. We can use the `ignoreVoid: true` flag to allow the explicit ignoring of promises. It's a bit of a change to make the switchover, but probably worth it. I'll look at it later today.